src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -131,6 +131,15 @@ public AssetProvider CreateAssetProvider(Checksum solutionChecksum, SolutionAsse
                 // private CTS token that inFlightSolution controls.
                 var solution = await solutionTask.WithCancellation(cancellationToken).ConfigureAwait(false);
 
+                // now that we've computed the solution, cache it to help out future requests.
+                using (await _gate.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
technically we could use a totally different lock here as this is really *solely* about atomatically reading/writing these tuples.  I wasn't certain if that would make things more confusing though having two locks.